src/test/java/com/puppycrawl/tools/checkstyle/internal/ImmutabilityTest.java

```diff
@@ -0,0 +1,467 @@
+         */
+        private static boolean isModuleProperty(JavaField javaField) {
+            boolean result = false;
+            final JavaClass containingClass = javaField.getOwner();
+            final ModuleDetails moduleDetails = MODULE_DETAILS_MAP.get(
+                containingClass.getFullName());
+            if (moduleDetails != null) {
+                final List<ModulePropertyDetails> properties = moduleDetails.getProperties();
+                result = properties.stream()
+                    .map(ModulePropertyDetails::getName)
+                    .anyMatch(moduleName -> moduleName.equals(javaField.getName()));
+            }
+            return result;
+        }
+
+        @Override
+        public boolean apply(JavaField input) {
+            return isModuleProperty(input);
+        }
+    }
+
+    /**
+     * DescribedPredicate defining condition for a class to have immutable fields.
+     */
+    private static final class ImmutableFieldsPredicate extends DescribedPredicate<JavaClass> {
+        private ImmutableFieldsPredicate() {
+            super("immutable fields");
+        }
+
+        @Override
+        public boolean apply(JavaClass input) {
+            final Set<JavaField> fields = input.getFields();
+            return fields.stream()
+                .filter(javaField -> {
+                    return !ModulePropertyPredicate.isModuleProperty(javaField)
+                        && !SUPPRESSED_FIELDS_IN_MODULES.contains(javaField.getFullName());
+                })
+                .allMatch(javaField -> {
+                    final Set<JavaModifier> javaFieldModifiers = javaField.getModifiers();
+                    boolean result = javaFieldModifiers.contains(JavaModifier.FINAL)
+                        && ImmutableFieldArchCondition.isImmutable(javaField);
+                    if (!javaField.getName().contains("MSG")) {
+                        result = result && javaFieldModifiers.contains(JavaModifier.PRIVATE);
+                    }
+                    return result;
+                });
+        }
+    }
+
+    /**
+     * ArchCondition checking if a class is present in the suppression list.
+     */
+    private static final class JavaClassSuppressionArchCondition extends ArchCondition<JavaClass> {
+
+        private final Set<String> suppressedClasses;
+
+        private JavaClassSuppressionArchCondition(Set<String> suppressedClasses,
+                                                  String description) {
+            super(description);
+            this.suppressedClasses = suppressedClasses;
+        }
+
+        @Override
+        public void check(JavaClass item, ConditionEvents events) {
+            if (!suppressedClasses.contains(item.getFullName())) {
+                final String message = String.format(
+                    Locale.ROOT, "should %s or resolved.", getDescription());
+                events.add(SimpleConditionEvent.violated(item, message));
+            }
+        }
+    }
```

Comment:
This will allow you to have only one condition class:

```suggestion
    private static final class ArchRuleSuppressionCondition<T extends HasName.AndFullName>
            extends ArchCondition<T> {

        private final Set<String> suppressedClasses;

        private ArchRuleSuppressionCondition(Set<String> suppressedClasses,
                                             String description) {
            super(description);
            this.suppressedClasses = suppressedClasses;
        }

        @Override
        public void check(T item, ConditionEvents events) {
            if (!suppressedClasses.contains(item.getFullName())) {
                final String message = String.format(
                    Locale.ROOT, "should %s or resolved.", getDescription());
                events.add(SimpleConditionEvent.violated(item, message));
            }
        }
    }
```